### ISSUE TITLE
mt6771 umidigi z2 pro, AOSP 9.0 v111

### ISSUE THREAD

**Issue by 2pferdebert** [phhusson/treble_experimentations#424]
first of all, thanks for your great work. Without your work I would have bought another phone.

everything works so far (not tested yet nfc), except SMS, Volte and Vowifi.

How can I help you to help me. (logs, bountys, testing, ...)

I'm not able to receive SMS and I can't send SMS either (I can dial a number, but the app doesn't let me enter massage "Failed to create conversation, please try again later").

 for volte and vowifi I don't get an option to choose.

**Comment by phhusson** [comment]
that's on stock unmodified boot.img?

**Comment by 2pferdebert** [comment]
everything is unmodified except the system image

**Comment by phhusson** [comment]
can you try another sms app?
Try forcing 2g?That's with one sim not 2?

**Comment by 2pferdebert** [comment]
I only got one sim card in slot 1, 

now i tested "sms/mms" app and and "messageges", with slot 1 and with 2g, 3g and 4g.
same result (no sms). 
and changed the card to slot 2 -» no mobil data, no sms, calls are working.

**Comment by phhusson** [comment]
k
Get me a logcat -b all when receiving sms then

**Comment by 2pferdebert** [comment]
[logUmidigiZ2Pro.txt](https://github.com/phhusson/treble_experimentations/files/2938705/logUmidigiZ2Pro.txt 

thank you for helping me.  i appreciate it very much.

**Comment by phhusson** [comment]
03-07 00:00:18.827  2527  2950 E GsmInboundSmsHandler: android.database.sqlite.SQLiteException: Can't downgrade database from version 661000 to 67

sounds like you failed at doing a factory reset...? Perhaps you tried to backup/restore your SMS?
Anyway, try to do adb root; adb shell pm clear com.android.internal.telephony and reboot

**Comment by 2pferdebert** [comment]
~/Downloads >>> adb root                                                                                                                                                       
restarting adbd as root
~/Downloads >>> adb shell pm clear com.android.internal.telephony                                                                                                              
Failed

tried also:
~/Downloads >>> adb shell 
255|phhgsi_arm64_a:/ # pm clear com.android.internal.telephony                                                                                                                 
Failed

and:
1|phhgsi_arm64_a:/ # su
:/data/data # pm clear com.android.internal.telephony
Failed

after this i did a full reset and now i can send sms :) but still cant receive :(. (but who needs sms nowadays ;-) )
thank you

i send a message (worked) and tried to receive one again but it also failed.
[logUmidigiZ2Pro.log](https://github.com/phhusson/treble_experimentations/files/2938813/logUmidigiZ2Pro.log)

**Comment by phhusson** [comment]
can you paste the result of lshal command?

Le jeu. 7 mars 2019 à 01:00, 2pferdebert <notifications@github.com> a
écrit :

**Comment by 2pferdebert** [comment]
voila 
[lshal.txt](https://github.com/phhusson/treble_experimentations/files/2938852/lshal.txt)

**Comment by phhusson** [comment]
hum. try, as root, to run:
/system/bin/mtk-sms-fwk-ready slot1
/system/bin/mtk-sms-fwk-ready slot2
Perhaps multiple times...
(don't reboot)

**Comment by 2pferdebert** [comment]
got an sms on the first time

**Comment by 2pferdebert** [comment]
and the other ones followes :+1:

**Comment by phhusson** [comment]
Could you reboot, wait like 3 minutes, and then an SMS and see if SMS are still coming?

**Comment by 2pferdebert** [comment]
Works even after the reboot

**Comment by 2pferdebert** [comment]
I also tested the second SIM card slot. SMS and 4g works also

**Comment by 2pferdebert** [comment]
Thank you Pierre.
Now the only missing functions are the voice over LTE/WiFi calls. 

And of course the bad WiFi itself. but this is a umidigi and mtk problem in the vendor partition, [I do not hope so], >>sometimes the WiFi breaks down for few seconds and works again for some minutes

**Comment by apogen** [comment]
Hi, could you please check if outgoing calls working in sim2?? 
thanks...

**Comment by 2pferdebert** [comment]
outgoing calls on slot 2 do not work for me.

after hotpluging the simcard slot 1 do also not work. after a reboot everything works again on slot 1. i tested also slot 2 with reboot but i got the same result. sms is working, incoming call also but not the outgoing call.
sorry i didnt tested it the days before

**Comment by ironashram** [comment]
/system/bin/mtk-sms-fwk-ready slot1
/system/bin/mtk-sms-fwk-ready slot2

those commands fixed my incoming sms issue on mt6771

**Comment by apogen** [comment]
yes.. incoming sms its ok now on both slots.. the problem now (at least for me) is no outgoing calls in slot2... a fix for this??

**Comment by 2pferdebert** [comment]
After some time the mobile connection( data, SMS, calls) is lost and I have to force a reconnection (flight modus on/off). And 3 times the reconnection was only possible after a reboot.
Most of the time it works perfect for me. ( Besides Voice over ... And outgoing calls on slot 2)

**Comment by chicato83** [comment]
Where can I download that ROM?

**Comment by WebPageKeySender** [comment]
> 
> 
> Where can I download that ROM?

Download latest GSI works, and you need to fix the sms gateway with the commands above.

**Comment by JonnyXDA** [comment]
> > Where can I download that ROM?
> 
> Download latest GSI....it works, and you need to fix the sms gateway with the commands above.

Does outgoing calls work on the latest GSI?

**Comment by WebPageKeySender** [comment]
> 
> 
> > > Where can I download that ROM?
> > 
> > 
> > Download latest GSI....it works, and you need to fix the sms gateway with the commands above.
> 
> Does outgoing calls work on the latest GSI?

yes i can call anyone

**Comment by chicato83** [comment]
> > Where can I download that ROM?
> 
> Download latest GSI....it works, and you need to fix the sms gateway with the commands above.

Can you put the link for download here?

**Comment by MAhmeed24** [comment]
I downloaded the last GSI but I have the problem of outgoing calls.  Could you help me?

**Comment by PhillipCF** [comment]
Can anyone help me install android 9 treble on my umidigi z2 pro? I've tried it in many ways but it always goes into the infinite loop. 
Twrp = 0mb internal storage, I format the data but according to the tutorial needs to wipe some partitions like vendor before installing rom (but vendor does not appear for wipe), in installing image, system partition does not appear. Install gsi by odb bricks the phone, By flash sp tools (system image only) also bricks. What am i doing wrong?

I'm using twrp by ghost
